OSX compilation issues for C++11

For OSX to compile C++ 11

- `clang++` needs to be used instead of `g++`
- This flag needs to be added to the compiler: `-std=gnu++11`

I wasn't sure how to safely make those changes, so if somebody else could for me then I can try compiling.

Well I was hoping to learn something new (how to work with configure and Makefiles) but I can't figure this thing out.

First tried to add the macro, then gave up and just tried to add `-std=c++11` to CFLAGS. But now after I run `autoconf` to regenerate the `configure` file, when I type `make` it gives this error:

    CDPATH="${ZSH_VERSION+.}:" && cd . && aclocal-1.15 
    /bin/sh: aclocal-1.15: command not found
    make: *** [aclocal.m4] Error 127

Even checking out the repository and starting over doesn't help, I have to remove the entire folder and clone the repo again.

--- a/configure
+++ b/configure
@@ -5354,7 +5354,7 @@ fi
 localstatedir=/var/lib
 CFLAGS="$CFLAGS $SDL_CFLAGS "
 LIBS="$LIBS $SDL_LIBS $SDL2_LIBS"
-CPPFLAGS="$SDL_CFLAGS $SDL2_CFLAGS -DLINUX -DLOCAL_STATE_DIR=\\\"$localstatedir/ivan\\\" -DDATADIR=\\\"$datadir\\\" -DUSE_SDL -DGCC"
+CPPFLAGS="$SDL_CFLAGS $SDL2_CFLAGS -DLINUX -DLOCAL_STATE_DIR=\\\"$localstatedir/ivan\\\" -DDATADIR=\\\"$datadir\\\" -DUSE_SDL -DGCC -std=c++11"
 
 
 

--- a/configure.ac
+++ b/configure.ac
@@ -30,7 +30,7 @@ fi
 localstatedir=/var/lib
 CFLAGS="$CFLAGS $SDL_CFLAGS "
 LIBS="$LIBS $SDL_LIBS $SDL2_LIBS"
-CPPFLAGS="$SDL_CFLAGS $SDL2_CFLAGS -DLINUX -DLOCAL_STATE_DIR=\\\"$localstatedir/ivan\\\" -DDATADIR=\\\"$datadir\\\" -DUSE_SDL -DGCC"
+CPPFLAGS="$SDL_CFLAGS $SDL2_CFLAGS -DLINUX -DLOCAL_STATE_DIR=\\\"$localstatedir/ivan\\\" -DDATADIR=\\\"$datadir\\\" -DUSE_SDL -DGCC -std=c++11"
 
 AC_PROG_INSTALL
 

